### ISSUE TITLE
Efficient distributed forward model

### ISSUE THREAD

**Issue by uellue** [Ptychography-4-0/ptychography#15]
Solvers require calculating a next sample vector from evaluating the error and/or local gradient of the forward model with respect to the measured data.

In LiberTEM, the data and computation can be distributed, parallelized and serialized as desired. Using this approach for solvers requires a forward model that can be distributed in the same fashion so that parts of the sample can be evaluated separately and the new sample vector resp. delta is merged from partial sample vector results.

**Comment by sk1p** [comment]
Is this something where we could plug in [diffsims](https://github.com/pyxem/diffsims)?

**Comment by uellue** [comment]
Yes, good point! For sure if we don't have to provide the Hessian and Jacobian for an optimizer. It would be interesting to see if diffsims can do that too or if it can/should be added.
